src/emqtt.erl

```diff
@@ -858,6 +872,7 @@ waiting_for_connack(cast, ?CONNACK_PACKET(?RC_SUCCESS,
     State2 = qoe_inject(connected, State1),
     State3 = ensure_retry_timer(ensure_keepalive_timer(State2)),
     Retry = [{next_event, info, immediate_retry} || not emqtt_inflight:is_empty(Inflight)],
+    ok = eval_msg_handler(State3, connected, Properties),
```

Comment:
Some tests are now failing (somewhere [here](https://github.com/emqx/emqtt/blob/952744480a1b6a516b5cdfb3730cfe21df857848/src/emqtt.erl#L1608)) because of this added line.